public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -222,6 +222,14 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
         return this.runQuery(options, target);
       });
 
+    // No valid targets, return the empty result to save a round trip.
+    if (_.isEmpty(subQueries)) {
```

Comment:
Small nitpick maybe try to use something else then lodash? And if not try to import only the `isEmpty` and not the whole `lodash` with `_`?

Reply:
Refactored. 🙂Also, lodash was already imported in the file. But I've imported both `isEmpty` and `isString` instead of whole lodash.

Reply:
Also, the same function is used in other datasources, so I was trying to keep it consistent.

Reply:
Great work!